docs/Users/Tutorials/Testing_Bears.rst

```diff
@@ -0,0 +1,158 @@
+How to use LocalBearTestHelper to test your bears
+=================================================
+
+*coala* has an awesome testing framework so you can write tests for bears
+with ease.
+
+You can use the following to test your bears:
+
+- ``LocalBearTestHelper.check_validity``
+- ``LocalBearTestHelper.check_results``
+- ``verify_local_bears``
+
+Understanding through examples
+------------------------------
+
+Let us understand how to write tests for ``MyBear`` written for
+``python``.
+
+.. code::
+
+    from coalib.results.Result import Result
+    from coalib.bears.LocalBear import LocalBear
+
+
+    class MyBear(LocalBear):
+
+        def run(file,filename):
+            if(len(file)>10):
+                yield Result(self, "Too many lines")
+
+**EXAMPLE 1** using ``verify_local_bear``
+
+.. code::
+
+    from bears.python.MyBear import MyBear
+    from tests.LocalBearTestHelper import verify_local_bear
+
+
+    good_file = '''
+
+    def HelloWorld():
+        print ('HelloWorld')
+
+    def main():
+        HelloWorld()
+
+    '''.splitlines(keepends=True)
+
+
+    bad_file = '''
+
+    def Hello():
+        return 'Hello'
+
+    def World():
+        return 'World'
+
+    def HelloWorld():
+        return Hello() + World()
+
+    def main():
+        HelloWorld()
+
+    '''.splitlines(keepends=True)
+
+    MyBearTest = verify_local_bear(MyBear,
+                                   valid_files=(good_file,),
+                                   invalid_files=(bad_file,))
+
+``good_file`` is a file which your bear considers as non-style-violating
+and a ``bad_file`` is one which has at least one error/warning/info.
+``My Bear`` checks if a file has less than or equal to 10 lines. So you need
```

Comment:
you can just move this above, so people can know about the bear before reading the example